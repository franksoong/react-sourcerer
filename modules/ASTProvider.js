import React from 'react';
import {parse} from 'babylon';

class ASTProvider extends React.Component {
  getChildContext() {
    return {
      ast: parse(this.props.source),
    };
  }

  render() {
    return (
      React.Children.only(this.props.children)
    );
  }
}

ASTProvider.childContextTypes = {
  ast: React.PropTypes.object,
};

export default ASTProvider;
