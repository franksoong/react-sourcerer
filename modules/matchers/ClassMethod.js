import React from 'react';
import Match from '../Match';
import * as t from 'babel-types';

class ClassMethod extends React.Component {
  render() {
    return (
      <Match
        {...this.props}
        match={(path) => {
          const {name} = this.props;
          const isClassMethod = t.isClassMethod(path);
          if (!isClassMethod) return false;
          if (name && path.node.computed) return false;
          if (name && name !== path.node.key.name) return false;
          return isClassMethod;
        }}
      />
    );
  }
}

export default ClassMethod;
