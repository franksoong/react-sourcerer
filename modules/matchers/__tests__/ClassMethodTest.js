import ASTProvider from '../../ASTProvider';
import React from 'react';
import ClassMethod from '../ClassMethod';

import renderer from 'react-test-renderer';
describe('ClassMethod', () => {
  it('should match a class method', () => {
    const output = renderer.create(<ASTProvider source={`
      class Person {
        greet() {
        }

        goodbye() {
        }
      }
    `}>
      <ClassMethod>
        {
          ((matches) => {
            return (
              <div>
                {matches.map((path) => path.node.type)}
              </div>
            );
          })
        }
      </ClassMethod>
    </ASTProvider>);

    expect(output.toJSON()).toMatchSnapshot(); 
  });

  it('should allow filtering based on method name', () => {
    const output = renderer.create(<ASTProvider source={`
      class Person {
        goodbye() {
        }

        greet() {
        }
      }
    `}>
      <ClassMethod name='greet'>
        {
          ((matches) => {
            return (
              <div>
                {matches.map((path) => path.node.key.name)}
              </div>
            );
          })
        }
      </ClassMethod>
    </ASTProvider>);

    expect(output.toJSON()).toMatchSnapshot(); 
  });

  it('should ignore computed method names', () => {
    const output = renderer.create(<ASTProvider source={`
      const computed = () => 'age'; 
      class Person {
        [goodbye]() {
        }

        greet() {
        }

        goodbye() {
        }
      }
    `}>
    <ClassMethod name='goodbye'>
        {
          ((matches) => {
            return (
              <div>
                {matches.map((path) => path.node.key.name)}
              </div>
            );
          })
        }
      </ClassMethod>
    </ASTProvider>);

    expect(output.toJSON()).toMatchSnapshot(); 
  });
});
