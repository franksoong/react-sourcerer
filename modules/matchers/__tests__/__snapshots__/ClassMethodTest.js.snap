exports[`ClassMethod should allow filtering based on method name 1`] = `
<div>
  greet
</div>
`;

exports[`ClassMethod should ignore computed method names 1`] = `
<div>
  goodbye
</div>
`;

exports[`ClassMethod should match a class method 1`] = `
<div>
  ClassMethod
  ClassMethod
</div>
`;
